libpermian/plugins/kickstart_test/__init__.py

```diff
@@ -151,15 +161,94 @@ def __init__(self, testRuns, crcList, arch):
         self.runner_command = self.settings.get('kickstart_test', 'runner_command').split()
         self.ksrepo = self.settings.get('kickstart_test', 'kstest_repo')
         self.ksrepo_branch = self.settings.get('kickstart_test', 'kstest_repo_branch')
+        self.retry = self.settings.getboolean('kickstart_test', 'retry_on_failure')
+        self.timeout = self.settings.get('kickstart_test', 'timeout')
+
+    def _create_defaults_file(self, content):
+        with tempfile.NamedTemporaryFile("w", delete=False, prefix="defaults-") as f:
+            f.write(content)
+            fpath = f.name
+        os.chmod(fpath, os.stat(fpath).st_mode | stat.S_IROTH)
+        return fpath
+
+    def _get_url_overrides(self, urls):
+        try:
+            arch_urls = urls[self.arch]
+        except KeyError:
+            return ""
+        kstest_var_map = {
+            'KSTEST_URL': 'installation_tree',
+            'KSTEST_METALINK': 'metalink',
+            'KSTEST_MIRRORLIST': 'mirrorlist',
+            'KSTEST_FTP_URL': 'ftp_url',
+            'KSTEST_MODULAR_URL': 'modular_url',
+        }
+        overrides = []
+        for variable, event_key in kstest_var_map.items():
+            value = arch_urls.get(event_key)
+            if value:
+                overrides.append(f"export {variable}={value}")
+
+        return "\n".join(overrides)
+
+    def process_installation_urls(self, urls):
+        """Get test run parameters from installationUrls event structure
+
+        :param urls: structure holding scenario data
+        :type urls: InstallationUrlsStructure
+        :returns: tuple with
+                  - boot.iso URL based on installation_tree location or None if
+                    the installation_tree location is not available
+                  - path of override defaults file with urls to be used by launcher
+                    or None if there are no relevant overrides
+        :rtype: (str, str))
+        """
+        boot_iso_url = None
+        defaults_file_path = None
+
+        try:
+            tree = urls[self.arch]['installation_tree']
+        except KeyError:
+            pass
+        else:
+            if tree:
+                boot_iso_url = os.path.join(tree, BOOT_ISO_PATH_IN_INSTALLATION_TREE)
+
+        # Configure installation repositories
+        variable_overrides = self._get_url_overrides(urls)
+        if variable_overrides:
+            defaults_file_path = self._create_defaults_file(variable_overrides)
+            LOGGER.info("Created override defaults file "
+                        f"{defaults_file_path} with content:"
+                        f"\n{variable_overrides}")
+
+        return (boot_iso_url, defaults_file_path)
 
     def setup(self):
         if self.arch not in SUPPORTED_ARCHITECTURES:
             LOGGER.info(f"Architecture {self.arch} is not supported.")
             raise UnsupportedConfiguration('architecture', self.arch)
 
+        if self.event.kstestParams:
+            try:
```

Comment:
platform must be in KstestParamsStructure, so this is not necessary

Reply:
Updated